### PR TITLE
chore(updatecli) fix UBI9 manifest 

### PR DESCRIPTION
**Pull request by gounthar**
This is a follow-up PR to #906 by @dduportal.
I started from [his work](https://github.com/jenkinsci/docker-agent/pull/906/files#diff-02266124aa9dfc148fb3f492952ca4add1a7d498a00e8ec0bec0699ec52d0bcbR20) and altered the `jq` query.

### Testing done

`updatecli diff --debug --config ./updatecli/updatecli.d/rhel-ubi9.yaml --values ./updatecli/values.github-action.yaml 2>&1`

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by dduportal — APPROVED** [review]

File: updatecli/scripts/ubi9-latest-tag.sh [hunk 1]
```diff
@@ -39,4 +39,13 @@ unique_tag=$(echo "$latest_tag" | sort | uniq | grep -v latest | grep "-")
 
 # Trim spaces
 unique_tag=$(echo "$unique_tag" | xargs)
-echo $unique_tag
+
+# Check if the latest_version is empty
+if [ -z "$unique_tag" ]; then
+  >&2 echo "Error: No valid version found for the 'latest' tag."
+  exit 1
+fi
```
Comment (lemeurherveCB) [inline updatecli/scripts/ubi9-latest-tag.sh:47]:
```suggestion
```
issue(non-blocking): already tested above.

Comment (lemeurherveCB) [reply updatecli/scripts/ubi9-latest-tag.sh:47]: same suggestion as https://github.com/jenkinsci/docker-agent/pull/906#discussion_r1860863386: simplify the script to pipe a trim on the unique result from the `curl | jq` command then only test if it's empty before `exit`ing 0.

File: updatecli/scripts/ubi9-latest-tag.sh [hunk 2]
```diff
@@ -39,4 +39,7 @@ unique_tag=$(echo "$latest_tag" | sort | uniq | grep -v latest | grep "-")
 
 # Trim spaces
 unique_tag=$(echo "$unique_tag" | xargs)
```
Comment (lemeurherveCB) [inline updatecli/scripts/ubi9-latest-tag.sh:41]: same (non-blocking) suggestion as https://github.com/jenkinsci/docker-agent/pull/906#discussion_r1860863386: simplify the script to pipe a trim on the unique result from the `curl | jq` command then only test if it's empty before `exit`ing 0.

(reposting as the previous discussion has been automatically hidden)

Comment (lemeurherveCB) [reply updatecli/scripts/ubi9-latest-tag.sh:41]: Then maybe your query can return multiple and duplicated tags? IDK

Comment (gounthar) [reply updatecli/scripts/ubi9-latest-tag.sh:41]: Yes, my query will return several tags, including `9.5` and `9.5-1731517889`, which will appear multiple times in the output. This repetition is typical of the data we are handling:

```
 "tags": [
            {
              "added_date": "2024-11-13T18:11:46.455000+00:00",
              "name": "9.5",
              "_links": {
                "tag_history": {
                  "href": "/v1/tag-history/registry/registry.access.redhat.com/repository/ubi9/tag/9.5"
                }
              }
            },
            {
              "added_date": "2024-11-13T18:11:46.455000+00:00",
              "name": "9.5-1731517889",
              "_links": {
                "tag_history": {
                  "href": "/v1/tag-history/registry/registry.access.redhat.com/repository/ubi9/tag/9.5-1731517889"
                }
              }
            },
            {
              "added_date": "2024-11-13T18:11:46.455000+00:00",
              "name": "latest",
              "_links": {
                "tag_history": {
                  "href": "/v1/tag-history/registry/registry.access.redhat.com/repository/ubi9/tag/latest"
                }
              }
            }
          ],
```

Comment (lemeurherveCB) [reply updatecli/scripts/ubi9-latest-tag.sh:41]: though: can this query be more precise to return only the latest tag?

Comment (gounthar) [reply updatecli/scripts/ubi9-latest-tag.sh:41]: Unfortunately, I'm not sure how to proceed with that task. 🤷 
Feel free to give it a try, as I'm not proficient with `jq`. 😉

**Review by dduportal — APPROVED** [review]
Approving and merging to fix the failing builds. Feel free to open a PR proposing a better change or to revert if it breaks anything.

**Comment by gounthar** [comment]
Thanks!

**Pull request merged**